pkg/asset/manifests/featuregate.go

```diff
@@ -0,0 +1,86 @@
+package manifests
+
+import (
+	"path/filepath"
+
+	"github.com/ghodss/yaml"
+	configv1 "github.com/openshift/api/config/v1"
+	"github.com/pkg/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/openshift/installer/pkg/asset"
+	"github.com/openshift/installer/pkg/asset/installconfig"
+)
+
+var fgFileName = filepath.Join(openshiftManifestDir, "99_feature-gate.yaml")
+
+// FeatureGate generates the feature gate manifest.
+type FeatureGate struct {
+	FileList []*asset.File
+	Config   configv1.FeatureGate
+}
+
+var _ asset.WritableAsset = (*Proxy)(nil)
+
+// Name returns a human-friendly name for the asset.
+func (*FeatureGate) Name() string {
+	return "Feature Gate Config"
+}
+
+// Dependencies returns all of the dependencies directly needed to generate
+// the asset.
+func (*FeatureGate) Dependencies() []asset.Asset {
+	return []asset.Asset{
+		&installconfig.InstallConfig{},
+	}
+}
+
+// Generate generates the FeatureGate CRD.
+func (f *FeatureGate) Generate(dependencies asset.Parents) error {
+	installConfig := &installconfig.InstallConfig{}
+	dependencies.Get(installConfig)
+
+	// A FeatureGate could be populated on every install,
+	// even for those using the default feature set, but for
+	// continuity let's only create a cluster feature gate
+	// when non-default feature gates are enabled.
```

Comment:
one potential wrinkle here is that implicit installer opinions will slip past #6247's opinion-collision guard.  That's probably fine in the "install-config didn't care" case, but may be surprising in the "install-config was explicitly...", oh, I guess [`Default` is the empty string][1].  So in all cases where the install-config set a non-empty string, we'll be creating this manifest, and getting opinion-collision guards.  Works for me :+1:

[1]: https://github.com/openshift/api/blob/66f8283afa3b5761a9c26f76137d074d0a43bfc9/config/v1/types_feature.go#L29-L30

Reply:
> So in all cases where the install-config set a non-empty string, we'll be creating this manifest, and getting opinion-collision guards. 

Correct

> Works for me +1 But creating the manifest every time would work for me too ;)

Yeah I debated this (which is why I included the comment). My opinion is pretty weakly held (i.e. I think either option is fine). I landed on not writing the manifest for the default feature set for two reasons:

1. maintain consistency
2. I have an assumption, that may be mistaken, that the existence of the featuregate resource may be a signal in and of itself. I would hope people wouldn't do an `oc get featuregate` without doing a `describe` too, but I highly doubt anyone is out there creating featuregate resources for a default featureset. So my reasoning was that omitting it in this case would be less confusing.